Update q.rd

DamienAlbert opened the pull request:
I had to do those changes to be able to make `gavo imp q.rd` with the latest epntap2.rd-2.xml installed.

DamienAlbert committed:
Update q.rd

Il m'a fallu faire ces modifications pour pouvoir faire un gavo imp q.rd avec epntap2.rd-2.xml installé.

Erard commented:
Hello Damien

I'm not quite sure what you're up to…
If I read correctly, you've edited the q.rd of another service: the CRISM/MRO service in Jacobs Univ.
To handle you're own service, you have to write a specific q.rd file adapted to your service/data. The examples here are only provided as examples. You can also use Mikhail's q.rd creator utility, at least to start with.

Cheers,
Stéphane 

Le 6 sept. 2017 à 17:00, DamienAlbert <notifications@github.com> a écrit :

> Il m'a fallu faire ces modifications pour pouvoir faire un gavo imp q.rd avec epntap2.rd-2.xml installé.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6
> 
> Commit Summary
> 
> 	• Update q.rd
> File Changes
> 
> 	• M mixin-EPN-TAP-2.0/RD/q.rd (5)
> Patch Links:
> 
> 	• https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6.patch
> 	• https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

DamienAlbert commented:
Sorry I was not very clear on my meaning : 

I was trying to make //epntap2 work for SSHADE, and saw an updated epntap2.rd in mixin-EPN-TAP-2.0 . After installing it following the readme, I tried to import a the testing q.rd found in mixin-EPN-TAP-2.0/RD by following the RD/readme.txt . It seems to be based on epn_vespa_crism with a small sample data.

It did not work on the first try because of some errors (maybe not errors, but those prevented the import to be successful in epntap2) . I had to do some minor modifications to be able to play the q.rd to the end.

Mainly : 
- updated `//epntap2#table` and `//epntap2#populate` to `//epntap2#table-2_0` and `//epntap2#populate-2_0` ; otherwise a warning was telling me that "table" was not found in "epntap2"
- moved `processing_level="3"` into a bind ; otherwise the error was that "processing_level" attribute was not expected in the mixin, and "processing_level" had no default value and must be in a bind.

I was not trying to create the SSHADE service at this point, but just play this "testing" q.rd through. 

Cheers,
Damien

europlanet-vespa commented:
OK, it may be that you're using an older version of the EPN-TAP mixin, or an older version of DACHS.
This is not very clear on my side either, I have two different behaviours on different machines (related to access_format).
As far as I know, the most recent is called epntap2.rd-2.xml

Mikhail and Pierre can explain what to install - but Pierre is working at a new update, to be certified by Markus in the coming weeks (will implement Spatial_Frame_Type = None, to be used in your case).
Maybe you can do test starting from another example - PDS_speclib and spectro_planets have just been installed in Paris (and they both contain spectra).


Stéphane 

Le 7 sept. 2017 à 10:41, DamienAlbert <notifications@github.com> a écrit :

> Sorry I was not very clear on my meaning :
> 
> I was trying to make //epntap2 work for SSHADE, and saw an updated epntap2.rd in mixin-EPN-TAP-2.0 . After installing it following the readme, I tried to import a the testing q.rd found in mixin-EPN-TAP-2.0/RD by following the RD/readme.txt . It seems to be based on epn_vespa_crism with a small sample data.
> 
> It did not work on the first try because of some errors (maybe not errors, but those prevented the import to be successful in epntap2) . I had to do some minor modifications to be able to play the q.rd to the end.
> 
> Mainly :
> 
> 	• updated //epntap2#table and //epntap2#populate to //epntap2#table-2_0 and //epntap2#populate-2_0 ; otherwise a warning was telling me that "table" was not found in "epntap2"
> 	• moved processing_level="3" into a bind ; otherwise the error was that "processing_level" attribute was not expected in the mixin, and "processing_level" had no default value and must be in a bind.
> I was not trying to create the SSHADE service at this point, but just play this "testing" q.rd through.
> 
> Cheers,
> Damien
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

lesidaner commented:
Hi

epntap2.rd is part of the dachs installation, you don't have anything to 
add, are we clear about that ?
You may update dachs if it's an old version
apt-get update; apt-get upgrade; gavo upgrade

regards
Pierre


On 07/09/2017 11:09, VESPA - EuroPlanet-RI wrote:
> OK, it may be that you're using an older version of the EPN-TAP mixin, 
> or an older version of DACHS.
> This is not very clear on my side either, I have two different 
> behaviours on different machines (related to access_format).
> As far as I know, the most recent is called epntap2.rd-2.xml
>
> Mikhail and Pierre can explain what to install - but Pierre is working 
> at a new update, to be certified by Markus in the coming weeks (will 
> implement Spatial_Frame_Type = None, to be used in your case).
> Maybe you can do test starting from another example - PDS_speclib and 
> spectro_planets have just been installed in Paris (and they both 
> contain spectra).
>
>
> Stéphane
>
> Le 7 sept. 2017 à 10:41, DamienAlbert <notifications@github.com> a écrit :
>
> > Sorry I was not very clear on my meaning :
> >
> > I was trying to make //epntap2 work for SSHADE, and saw an updated 
> epntap2.rd in mixin-EPN-TAP-2.0 . After installing it following the 
> readme, I tried to import a the testing q.rd found in 
> mixin-EPN-TAP-2.0/RD by following the RD/readme.txt . It seems to be 
> based on epn_vespa_crism with a small sample data.
> >
> > It did not work on the first try because of some errors (maybe not 
> errors, but those prevented the import to be successful in epntap2) . 
> I had to do some minor modifications to be able to play the q.rd to 
> the end.
> >
> > Mainly :
> >
> > • updated //epntap2#table and //epntap2#populate to 
> //epntap2#table-2_0 and //epntap2#populate-2_0 ; otherwise a warning 
> was telling me that "table" was not found in "epntap2"
> > • moved processing_level="3" into a bind ; otherwise the error was 
> that "processing_level" attribute was not expected in the mixin, and 
> "processing_level" had no default value and must be in a bind.
> > I was not trying to create the SSHADE service at this point, but 
> just play this "testing" q.rd through.
> >
> > Cheers,
> > Damien
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6#issuecomment-327740175>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAjyUHskIRcu5CHHt0R0ccTAvQphC0a2ks5sf7KygaJpZM4POgtn>.
>

-- 
-------------------------------------------------------------------------
                            Pierre Le Sidaner
                         Observatoire de Paris

Direction Informatique de l'Observatoire
Observatoire Virtuel 01 40 51 20 82
61, avenue de l'Observatoire 75014 Paris

mailto:pierre.lesidaner@obspm.fr
http://padc.obspm.fr

--------------------------------------------------------------------------

BaptisteCecconi commented:
Dear all,

I would recommend to always start with the latest version of DaCHS and use the embedded epntap2.rd mixin. We should make sure that the epntap2.rd mixin proposed on github (by Mihail) is up-to-date, and we should also mention, what features it adds on top of the official epntap2.rd in the DaCHS distribution.

baptiste


> Le 7 sept. 2017 à 10:09, VESPA - EuroPlanet-RI <notifications@github.com> a écrit :
> 
> OK, it may be that you're using an older version of the EPN-TAP mixin, or an older version of DACHS.
> This is not very clear on my side either, I have two different behaviours on different machines (related to access_format).
> As far as I know, the most recent is called epntap2.rd-2.xml
> 
> Mikhail and Pierre can explain what to install - but Pierre is working at a new update, to be certified by Markus in the coming weeks (will implement Spatial_Frame_Type = None, to be used in your case).
> Maybe you can do test starting from another example - PDS_speclib and spectro_planets have just been installed in Paris (and they both contain spectra).
> 
> 
> Stéphane 
> 
> Le 7 sept. 2017 à 10:41, DamienAlbert <notifications@github.com> a écrit :
> 
> > Sorry I was not very clear on my meaning :
> > 
> > I was trying to make //epntap2 work for SSHADE, and saw an updated epntap2.rd in mixin-EPN-TAP-2.0 . After installing it following the readme, I tried to import a the testing q.rd found in mixin-EPN-TAP-2.0/RD by following the RD/readme.txt . It seems to be based on epn_vespa_crism with a small sample data.
> > 
> > It did not work on the first try because of some errors (maybe not errors, but those prevented the import to be successful in epntap2) . I had to do some minor modifications to be able to play the q.rd to the end.
> > 
> > Mainly :
> > 
> > • updated //epntap2#table and //epntap2#populate to //epntap2#table-2_0 and //epntap2#populate-2_0 ; otherwise a warning was telling me that "table" was not found in "epntap2"
> > • moved processing_level="3" into a bind ; otherwise the error was that "processing_level" attribute was not expected in the mixin, and "processing_level" had no default value and must be in a bind.
> > I was not trying to create the SSHADE service at this point, but just play this "testing" q.rd through.
> > 
> > Cheers,
> > Damien
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6#issuecomment-327740175>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AL633V4TbsuScCIznK27m4hc-nwHpb-nks5sf7KygaJpZM4POgtn>.
>

europlanet-vespa commented:
In fact, adding version tracking in the mixin seems essential - at least the date and possibly a comment on variations implemented.

Stéphane 

Le 7 sept. 2017 à 11:19, Baptiste Cecconi <notifications@github.com> a écrit :

> Dear all,
> 
> I would recommend to always start with the latest version of DaCHS and use the embedded epntap2.rd mixin. We should make sure that the epntap2.rd mixin proposed on github (by Mihail) is up-to-date, and we should also mention, what features it adds on top of the official epntap2.rd in the DaCHS distribution.
> 
> baptiste
> 
> 
> > Le 7 sept. 2017 à 10:09, VESPA - EuroPlanet-RI <notifications@github.com> a écrit :
> > 
> > OK, it may be that you're using an older version of the EPN-TAP mixin, or an older version of DACHS.
> > This is not very clear on my side either, I have two different behaviours on different machines (related to access_format).
> > As far as I know, the most recent is called epntap2.rd-2.xml
> > 
> > Mikhail and Pierre can explain what to install - but Pierre is working at a new update, to be certified by Markus in the coming weeks (will implement Spatial_Frame_Type = None, to be used in your case).
> > Maybe you can do test starting from another example - PDS_speclib and spectro_planets have just been installed in Paris (and they both contain spectra).
> > 
> > 
> > Stéphane 
> > 
> > Le 7 sept. 2017 à 10:41, DamienAlbert <notifications@github.com> a écrit :
> > 
> > > Sorry I was not very clear on my meaning :
> > > 
> > > I was trying to make //epntap2 work for SSHADE, and saw an updated epntap2.rd in mixin-EPN-TAP-2.0 . After installing it following the readme, I tried to import a the testing q.rd found in mixin-EPN-TAP-2.0/RD by following the RD/readme.txt . It seems to be based on epn_vespa_crism with a small sample data.
> > > 
> > > It did not work on the first try because of some errors (maybe not errors, but those prevented the import to be successful in epntap2) . I had to do some minor modifications to be able to play the q.rd to the end.
> > > 
> > > Mainly :
> > > 
> > > • updated //epntap2#table and //epntap2#populate to //epntap2#table-2_0 and //epntap2#populate-2_0 ; otherwise a warning was telling me that "table" was not found in "epntap2"
> > > • moved processing_level="3" into a bind ; otherwise the error was that "processing_level" attribute was not expected in the mixin, and "processing_level" had no default value and must be in a bind.
> > > I was not trying to create the SSHADE service at this point, but just play this "testing" q.rd through.
> > > 
> > > Cheers,
> > > Damien
> > > 
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub, or mute the thread.
> > > 
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub <https://github.com/epn-vespa/DaCHS-for-VESPA/pull/6#issuecomment-327740175>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AL633V4TbsuScCIznK27m4hc-nwHpb-nks5sf7KygaJpZM4POgtn>.
> > 
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

DamienAlbert commented:
> Maybe you can do test starting from another example - PDS_speclib

In fact that was the first thing I tried. But with my (updated) installation of gavo, I had the following error : 

```
$ gavo imp q.rd
*** Error: At or near /var/gavo/inputs/PDS_speclib/q.rd, (243, 4): At
/var/gavo/inputs/PDS_speclib/q.rd, (243, 4): May not bind non-existing
parameter(s) access_format.
```
This is why I tried to update my `__system__/epntap2.rd` with the `epntap2.rd-2.xml` one. With no success, as a new error is : 
```
$ gavo imp q.rd
Making data import
*X*X* Uncaught exception at toplevel
*** Error: Oops.  Unhandled exception ProgrammingError.

Exception payload: schema "pds_speclib" already exists
```

But anyway, if I revert the `__system__/epntap2.rd` to it's original released version, and leave the testing epn_vespa_crism `RD/q.rd` as is , I still have the bug with gavo unable to find the mixin //epntap2#table . ie. : 

```
$ gavo imp q.rd
*** Error: At or near /var/gavo/inputs/epn_vespa_crism/q.rd, (37, 16):
Element with id u'table' could not be located in RD __system__/epntap2
```

Which is why I made the modifications to //epntap2#table-2_0 mixin. The part with `processing_level` was just me trying to avoid the next errors returned by gavo imp.

I have not tried the generator, but I saw in the code that the mixin used was `//epntap2#table` which may not work.

I think the best way for me to advance is by copying the epn_vespa_crism q.rd as it works with the original epntap2.rd and a few modifications.

**TL;DR** :  
- With the original or new epntap2.rd , I cannot import the testing epn_vespa_crism q.rd , unless I modify the mixins to add `-2_0` to the end.
- With the original epntap2.rd I cannot import PDS_speclib because of the `access_format` attribute.
- With the new epntap2.rd-2 I cannot import PDS_speclib because of `schema "pds_speclib" already exists` error.

europlanet-vespa commented:
Stéphane 

Le 7 sept. 2017 à 11:52, DamienAlbert <notifications@github.com> a écrit :

> Maybe you can do test starting from another example - PDS_speclib
> 
> In fact that was the first thing I tried. But with my (updated) installation of gavo, I had the following error :
> 
> $ gavo imp q.rd
> *** Error: At or near /var/gavo/inputs/PDS_speclib/q.rd, (243, 4): At
> /var/gavo/inputs/PDS_speclib/q.rd, (243, 4): May not bind non-existing
> parameter(s) access_format.
> 

• OK, I've updated this one this morning, it should do now.

> This is why I tried to update my __system__/epntap2.rd with the epntap2.rd-2.xml one. With no success, as a new error is :
> 
> $ gavo imp q.rd
> Making data import
> *X*X* Uncaught exception at toplevel
> *** Error: Oops.  Unhandled exception ProgrammingError.
> 
> Exception payload: schema "pds_speclib" already exists
> 

• This looks like something different. Try to drop the previous version of the service.


> But anyway, if I revert the __system__/epntap2.rd to it's original released version, and leave the testing epn_vespa_crism RD/q.rd as is , I still have the bug with gavo unable to find the mixin //epntap2#table . ie. :
> 
> $ gavo imp q.rd
> *** Error: At or near /var/gavo/inputs/epn_vespa_crism/q.rd, (37, 16):
> Element with id u'table' could not be located in RD __system__/epntap2
> 
> Which is why I made the modifications to //epntap2#table-2_0 mixin. The part with processing_level was just me trying to avoid the next errors returned by gavo imp.
> 
> I have not tried the generator, but I saw in the code that the mixin used was //epntap2#table which may not work.

• As Baptiste mentioned, the mixin is part of DaCHS and must not be modified. 
But some examples were adapted to a previous version and may be obsolete.
In case of mismatch you should make sure you're using the latest mixin, and possibly change the q.rd - not the opposite.


> 
> I think the best way for me to advance is by copying the epn_vespa_crism q.rd as it works with the original epntap2.rd and a few modifications.
> 
> TL;DR :
> 
> 	• With the original or new epntap2.rd , I cannot import the testing epn_vespa_crism q.rd , unless I modify the mixins to add -2_0 to the end.
> 	• With the original epntap2.rd I cannot import PDS_speclib because of the access_format attribute.

• Try with this morning version, it should fix the access_format issue


> 	• With the new epntap2.rd-2 I cannot import PDS_speclib because of schema "pds_speclib" already exists error.

• Try to drop every possible past version of this service, either PDS_speclib and pds_speclib
then start again. Traces of the service persist even when you remove the files/directory, in which case you may have to redeclare the directory and drop the service from there (I think I've changed the case at some point).
+ always import the q.rd from the correct directory, and to not move anything afterwards (or first drop the service).

Stéphane

> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

DamienAlbert commented:
> OK, I've updated this one this morning, it should do now.

Allright, I had to drop PDS_speclib, lowercase the directory, drop db rows from various tables ; but at least pds_speclib is trying to import with the release version of epntap2.rd . 

There is still an error with `granule_gid` as it is not present in the csv but the rowmaker is : 
`<var key="granule_gid" source="granule_gid" />`. When no value is found, the error is then : 
```
$ gavo imp q.rd
Making data import
Starting /var/gavo/inputs/pds_speclib/data/spectrodb_ex.csv
Done /var/gavo/inputs/pds_speclib/data/spectrodb_ex.csv, read 1
*** Error: Row {...}
Field granule_gid: While assigning granule_gid in None: Key
'granule_gid' not found in a mapping.
```

>In case of mismatch you should make sure you're using the latest mixin

So must I install the new `epntap2.rd-2.xml` , or keep the original one ?

>, and possibly change the q.rd - not the opposite.

That's what I'm doing ; I'm only modifying the test epn_vespa_crism/q.rd because the mixin `//epntap2#table` does not exist. And to allow it to work I must also remove `processing_level` attribute, and comment the bindings processing_level, spatial_frame_type and access_format.

Once done, I can import it with the release epntap2.rd .

europlanet-vespa commented:
I think that's related to the example csv file, which is older than the q.rd. 
1) The name of the csv file must certainly be changed in the q.rd but I think you've done that
2) The actual file is too large to be stored there, and this excerpt is apparently outdated. Try with the one attached. If you got that far, I think this is OK anyway.
3) You may want to try and install spectro_planets, the csv contains all the dat and it should install properly
4) Be careful not to register your test services! That would be a mess, later.

Stéphane 




Le 7 sept. 2017 à 13:02, DamienAlbert <notifications@github.com> a écrit :

> OK, I've updated this one this morning, it should do now.
> 
> Allright, I had to drop PDS_speclib, lowercase the directory, drop db rows from various tables ; but at least pds_speclib is trying to import with the release version of epntap2.rd .
> 
> There is still an error with granule_gid as it is not present in the csv but the rowmaker is :
> <var key="granule_gid" source="granule_gid" />. When no value is found, the error is then :
> 
> $ gavo imp q.rd
> Making data import
> Starting /var/gavo/inputs/pds_speclib/data/spectrodb_ex.csv
> Done /var/gavo/inputs/pds_speclib/data/spectrodb_ex.csv, read 1
> *** Error: Row {...}
> Field granule_gid: While assigning granule_gid in None: Key
> 'granule_gid' not found in a mapping.
> 
> In case of mismatch you should make sure you're using the latest mixin
> 
> So must I install the new epntap2.rd-2.xml , or keep the original one ?
> 
> , and possibly change the q.rd - not the opposite.
> 
> That's what I'm doing ; I'm only modifying the test epn_vespa_crism/q.rd because the mixin //epntap2#table does not exist. And to allow it to work I must also remove processing_level attribute, and comment the bindings processing_level, spatial_frame_type and access_format.
> 
> Once done, I can import it with the release epntap2.rd .
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

DamienAlbert commented:
Ok, spectro_planets is working as intended (just had to rename `spectro_planet.csv` to `spectroplanet.csv` ).

Don't worry I won't try to register a service until much later. ^^
